core/src/main/java/org/apache/accumulo/core/client/admin/ActiveCompaction.java

```diff
@@ -132,27 +133,10 @@ public enum CompactionReason {
    */
   public abstract List<IteratorSetting> getIterators();
 
-  /**
-   * @since 2.1.0
-   */
-  public interface CompactionHost {
-    enum Type {
-      TSERVER, COMPACTOR
-    }
-
-    Type getType();
-
-    String getAddress();
-
-    int getPort();
-
-    String getResourceGroup();
-  }
-
   /**
    * Return the host where the compaction is running.
    *
    * @since 2.1.0
    */
-  public abstract CompactionHost getHost();
+  public abstract ServerId getHost();
```

Comment:
Could deprecate the existing method and introduce a new method all in 4.0.

```suggestion
  @Deprecated
  public abstract CompactionHost getHost();

  //@since 4.0.0  
  public abstract ServerId getServerId();
```